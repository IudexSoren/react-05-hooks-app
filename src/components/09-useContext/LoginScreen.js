import { useContext } from "react"
import { UserContext } from "./UserContext"

const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

  const handleClick = () => {
    const user = {
      id: 4444,
      name: 'Jhin'
    }
    setUser(user);
  }

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr/>
      <button
        className="btn btn-outline-dark"
        onClick={ handleClick }
      >Login</button>
    </div>
  )
}

export default LoginScreen
