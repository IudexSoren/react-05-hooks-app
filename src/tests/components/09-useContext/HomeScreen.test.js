import { mount } from 'enzyme';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas de HomeScreen', () => {

  const user = {
    name: 'Purrumaw',
    email: 'purru.maw@gmail.com'
  }

  const wrapper = mount(
    <UserContext.Provider value={{
      user
    }}>
      <HomeScreen />
    </UserContext.Provider>

  );

  test('Debe renderizar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });


})
