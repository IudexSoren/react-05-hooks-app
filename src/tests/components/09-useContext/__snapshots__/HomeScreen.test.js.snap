// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de HomeScreen Debe renderizar correctamente el componente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Purrumaw",
   "email": "purru.maw@gmail.com"
}
  </pre>
</div>
`;
